src/style/style_layer/fill_extrusion_style_layer.js

```diff
@@ -362,4 +362,16 @@ function fourSample(demSampler: DEMSampler, posX: number, posY: number, offsetX:
     ];
 }
 
+// Pretty much identical to gl-matrix's vc.transformMat4, except it assures w is > 0 before division.
+function transformMat4(out, a, m) {
+    const x = a[0],
+        y = a[1],
+        z = a[2];
+    const w = Math.max(m[3] * x + m[7] * y + m[11] * z + m[15], 0.00001);
+    out[0] = (m[0] * x + m[4] * y + m[8] * z + m[12]) / w;
+    out[1] = (m[1] * x + m[5] * y + m[9] * z + m[13]) / w;
+    out[2] = (m[2] * x + m[6] * y + m[10] * z + m[14]) / w;
+    return out;
+}
+
```

Comment:
I don't think you need a specialized transform function for this purpose. You could use a generic matrix multiplication instead (m44 * m41) and divide the resulting vector by w afterwards. Something like this:
```
const vec = mul(m, [a0, a1, a2, 1]);
vec /= max(vec[3], 0.00001);
```

Reply:
👍  https://github.com/mapbox/mapbox-gl-js/pull/10428/commits/451978987aa2fa3a4e944a7e37f2469f11fff2ac